### PR TITLE
Workaround for recent CMake bug w.r.t. CF guard

### PR DESCRIPTION
**Pull request by janvorli**
Recent versions of CMake generates the <LinkControlFlowGuard> node
under the <Link> node when it finds /guard:cf linker option in the
linker options. But that is not correct. It should generate it under
the <PropertyGroup> node so that the Microsoft.CppCommon.targets would
then generate <ControlFlowGuard> node under the <Link>.

To workaround the problem, change the casing of /guard:cf to uppercase.
CMake then doesn't detect the option as something known to it and passes
the option to the linker as additional option, which fixes the problem.

- Please add a description for changes you are making.
- If there is an issue related to this PR, please add a reference to it.
- If this PR should not run tests please add say skip_ci_please in this description replacing the underscores with spaces.

### CODE REVIEW

File: src/settings.cmake [hunk 1]
```diff
@@ -155,8 +155,8 @@ if(WIN32)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /DEBUG /PDBCOMPRESS")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /STACK:1572864")
 
-    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /guard:cf")
-    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /guard:cf")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /GUARD:CF")
```
Comment (gkhanna79) [inline src/settings.cmake]: Do we not need to fix https://github.com/dotnet/core-setup/pull/2654/files#diff-5c2eb952c0365ec0c860557785f5d26fR139?

Likewise for the other repos too.

Comment (janvorli) [inline src/settings.cmake]: We don't. The issue is only with the linker flag. And I have verified that all the target binaries contain the right cf guard stuff using dumpbin.

**Comment by janvorli** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by janvorli** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please

**Comment by sergiy-k** [comment]
@janvorli, is your PR ready?

**Comment by janvorli** [comment]
@sergiy-k oh, I have thought that it was merged in a long time ago.

**Pull request merged**